src/ai/backend/manager/sokovan/deployment/deployment_controller.py

```diff
@@ -75,14 +76,17 @@ async def create_deployment(
 
     async def _validate_model_revision(self, model_revision: ModelRevisionSpec) -> None:
         """Validate the model revision specification."""
-        definition_files = await self._deployment_repository.fetch_definition_files(
-            vfolder_id=model_revision.mounts.model_vfolder_id,
-            model_definition_path=model_revision.mounts.model_definition_path,
-        )
-        if definition_files.service_definition:
-            variant_def = ModelServiceDefinition.model_validate(definition_files.service_definition)
-            model_revision = variant_def.ovrride_model_revision(model_revision)
-        ModelDefinition.model_validate(definition_files.model_definition)
+        if model_revision.execution.runtime_variant == RuntimeVariant.CUSTOM:
+            definition_files = await self._deployment_repository.fetch_definition_files(
+                vfolder_id=model_revision.mounts.model_vfolder_id,
+                model_definition_path=model_revision.mounts.model_definition_path,
+            )
+            if definition_files.service_definition:
+                variant_def = ModelServiceDefinition.model_validate(
+                    definition_files.service_definition
+                )
+                model_revision = variant_def.override_model_revision(model_revision)
+            ModelDefinition.model_validate(definition_files.model_definition)
```

Comment:
It seems like the remaining type handling is also necessary, but has it not been applied yet?